### ISSUE TITLE
Mutating webhook for DELETE operations fail

### ISSUE THREAD

**Issue by tamalsaha** [kubernetes/kubernetes#66412]
We had mutating webhooks on delete operation. After updating to Kubernetes 1.11, it is failing.
Stacktrace:

```
1 status.go:64] apiserver received an error that is not an metav1.Status: &errors.errorString{s:"expected pointer, but got invalid kind"}

goroutine 9882 [running]:
runtime/debug.Stack(0xc4200b2008, 0xc42b3bdae8, 0x1)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/conversion.EnforcePtr(0x0, 0x0, 0xc42045a740, 0xc4201a6210, 0x6b58cd0b7ea986de, 0xc423ee1248, 0x1)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/conversion/helper.go:33 +0x159
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/conversion.(*Converter).doConversion(0xc420ac0700, 0x0, 0x0, 0x0, 0x0, 0x8, 0xc426845d40, 0xc42b3bdc88, 0x407533, 0xc42b3bdd68)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/conversion/converter.go:498 +0x50
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/conversion.(*Converter).Convert(0xc420ac0700, 0x0, 0x0, 0x0, 0x0, 0x8, 0xc426845d40, 0x40e8da4000000000, 0xc42b3bddc8)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/conversion/converter.go:480 +0xb6
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime.(*Scheme).Convert(0xc420820a20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc42b3bde78, 0x1)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime/scheme.go:484 +0x28a
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating.(*mutatingDispatcher).Dispatch(0xc4203d5680, 0x5127e80, 0xc4200ae018, 0xc4261bf3e0, 0xc4267b9710, 0x1, 0x1, 0x1, 0x8)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go:77 +0x47b
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/generic.(*Webhook).Dispatch(0xc420c35540, 0x5147c00, 0xc42ee9f860, 0x370a579b, 0x97306c891)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go:201 +0x39b
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating.(*Plugin).Admit(0xc420e15ee0, 0x5147c00, 0xc42ee9f860, 0x7f69b7d8db20, 0xc420e15ee0)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/plugin.go:95 +0x42
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics.pluginHandlerWithMetrics.Admit(0x50f0e40, 0xc420e15ee0, 0xc4203d5770, 0xc4203d5800, 0x1, 0x1, 0x5147c00, 0xc42ee9f860, 0x1, 0x7f69b7604ea8)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics/metrics.go:85 +0xca
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission.chainAdmissionHandler.Admit(0xc420363e00, 0xb, 0x10, 0x5147c00, 0xc42ee9f860, 0x370a1933, 0x973068a27)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/chain.go:35 +0xf5
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics.pluginHandlerWithMetrics.Admit(0x50f4620, 0xc420e76660, 0xc420390960, 0x0, 0x0, 0x0, 0x5147c00, 0xc42ee9f860, 0x1, 0x7f69b7604ea8)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics/metrics.go:85 +0xca
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission.auditHandler.Admit(0x50f0d80, 0xc420e752c0, 0x0, 0x5147c00, 0xc42ee9f860, 0xc4289f5501, 0x7f69b7604e40)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/audit.go:56 +0xf1
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.DeleteResource.func1(0x511dd80, 0xc4267b9620, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/delete.go:108 +0xdaf
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/metrics.InstrumentHandlerFunc.func1(0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:247 +0x1e0
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver.(*crdHandler).ServeHTTP(0xc420392580, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go:236 +0x4a2
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc429a43480, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:248 +0x266
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc4206641c0, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0x8a
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3c8ede7, 0x17, 0xc420af2000, 0xc4206641c0, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:154 +0x6a8
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc427060800, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:254 +0x3de
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc420664a10, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0x8a
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3c6ef0a, 0xe, 0xc4200cecf0, 0xc420664a10, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:154 +0x6a8
k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/apiserver.(*proxyHandler).ServeHTTP(0xc42d3224d0, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go:91 +0x169
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc42c7937c0, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:248 +0x266
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc425b96770, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0x8a
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3c71e64, 0xf, 0xc429a213b0, 0xc425b96770, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:154 +0x6a8
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authorization.go:64 +0x4bd
net/http.HandlerFunc.ServeHTTP(0xc42a7a08c0, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/usr/local/go/src/net/http/server.go:1947 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1(0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/maxinflight.go:160 +0x3fc
net/http.HandlerFunc.ServeHTTP(0xc42a79def0, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/usr/local/go/src/net/http/server.go:1947 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go:49 +0x1ce4
net/http.HandlerFunc.ServeHTTP(0xc42a7a0900, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/usr/local/go/src/net/http/server.go:1947 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0x7f69b7603ed0, 0xc4267b9618, 0xc4252dda00)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authentication.go:75 +0x2b3
net/http.HandlerFunc.ServeHTTP(0xc42a7a0940, 0x7f69b7603ed0, 0xc4267b9618, 0xc4252dd900)
	/usr/local/go/src/net/http/server.go:1947 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1(0xc42a797320, 0x5128f00, 0xc4267b9618, 0xc4252dd900, 0xc426a15080)
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:97 +0x8d
created by k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP
	/home/ac/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:96 +0x1b5
```

I tried to debug and this is what I found:

1. https://github.com/kubernetes/apiserver/blob/kubernetes-1.11.0/pkg/endpoints/handlers/delete.go#L106
2. https://github.com/kubernetes/apiserver/blob/kubernetes-1.11.0/pkg/admission/plugin/webhook/generic/webhook.go#L201
3. https://github.com/kubernetes/apiserver/blob/kubernetes-1.11.0/pkg/admission/plugin/webhook/mutating/dispatcher.go#L77
4. https://github.com/kubernetes/apimachinery/blob/kubernetes-1.11.0/pkg/runtime/scheme.go#L484
5. https://github.com/kubernetes/apimachinery/blob/kubernetes-1.11.0/pkg/conversion/converter.go#L498

`attr.Attributes.GetObject()` is nil. Calling `EnforcePtr(nil)` causes this error. I see that the code path has not changed since 1.10. But we don't see the same error in 1.10 minikube cluster. I can't explain that. 

We have stopped using mutating webhook on delete operation anyway, since it does not make much sense. But I thought I should file a bug just in case.

/kind bug
/sig api-machinery

**Comment by tamalsaha** [comment]
cc: @deads2k @sttts

**Comment by liggitt** [comment]
thanks for the report... is this on a custom resource or a native resource?

**Comment by tamalsaha** [comment]
It is a CRD.

**Comment by liggitt** [comment]
fix is in #66425, pick to 1.11.x in #66426

> But we don't see the same error in 1.10 minikube cluster. I can't explain that.

conversion of unstructured objects (used internally for CRDs) changed in 1.11 as part of CRD versioning, which made them actually go through the conversion stack
